source/fahrplanparser.d

```diff
@@ -80,78 +151,240 @@ class UnexpectedValueException(T) : Exception
-    assertThrown!DateTimeException("<dp><st><t>0061</t></st></dp>".readDocument.parseXPath("/dp")
-            .front.departureTime);
-    assertThrown!DateTimeException("<dp><st><t>2567</t></st></dp>".readDocument.parseXPath("/dp")
-            .front.departureTime);
-    assertThrown!DateTimeException("<dp><st><t></t></st></dp>".readDocument.parseXPath("/dp")
-            .front.departureTime);
-    assertThrown!DateTimeException("<dp><st><t>0</t></st></dp>".readDocument.parseXPath("/dp")
-            .front.departureTime);
-    assertThrown!DateTimeException("<dp><st><t>00</t></st></dp>".readDocument.parseXPath("/dp")
-            .front.departureTime);
-    assertThrown!DateTimeException("<dp><st><t>000000</t></st></dp>".readDocument.parseXPath("/dp")
-            .front.departureTime);
-    assertThrown!DateTimeException("<dp><st><t>00:00</t></st></dp>".readDocument.parseXPath("/dp")
-            .front.departureTime);
-    assertThrown!DateTimeException("<dp><st><t>abcd</t></st></dp>".readDocument.parseXPath("/dp")
-            .front.departureTime);
+@system unittest
+{
+    (// dfmt off
+    "<?xml version='1.0' encoding='UTF-8'?>\n" ~
+    "<dp>\n" ~
+    "    <st>\n" ~
+    "        <t>0</t>\n" ~
+    "    </st>\n" ~
+    "</dp>"
+    // dfmt on
+    ).parseDOM.children.filter!(node => node.name == departureNodeName)
+        .front.departureTime.should.throwException!DateTimeException;
+}
+
+@system unittest
+{
+    (// dfmt off
+    "<?xml version='1.0' encoding='UTF-8'?>\n" ~
+    "<dp>\n" ~
+    "    <st>\n" ~
+    "        <t>00</t>\n" ~
+    "    </st>\n" ~
+    "</dp>"
+    // dfmt on
+    ).parseDOM.children.filter!(node => node.name == departureNodeName)
+        .front.departureTime.should.throwException!DateTimeException;
+}
+
+@system unittest
+{
+    (// dfmt off
+    "<?xml version='1.0' encoding='UTF-8'?>\n" ~
+    "<dp>\n" ~
+    "    <st>\n" ~
+    "        <t>000000</t>\n" ~
+    "    </st>\n" ~
+    "</dp>"
+    // dfmt on
+    ).parseDOM.children.filter!(node => node.name == departureNodeName)
+        .front.departureTime.should.throwException!DateTimeException;
+}
+
+@system unittest
+{
+    (// dfmt off
+    "<?xml version='1.0' encoding='UTF-8'?>\n" ~
+    "<dp>\n" ~
+    "    <st>\n" ~
+    "        <t>00:00</t>\n" ~
+    "    </st>\n" ~
+    "</dp>"
+    // dfmt on
+    ).parseDOM.children.filter!(node => node.name == departureNodeName)
+        .front.departureTime.should.throwException!DateTimeException;
 }
 
-auto delay(XmlNode dp)
+@system unittest
+{
+    (// dfmt off
+    "<?xml version='1.0' encoding='UTF-8'?>\n" ~
+    "<dp>\n" ~
+    "    <st>\n" ~
+    "        <t>abcd</t>\n" ~
+    "    </st>\n" ~
+    "</dp>"
+    ).parseDOM.children.filter!(node => node.name == departureNodeName)
+        .front.departureTime.should.throwException!DateTimeException;
+}
+
+auto delay(DOMEntity!string dp)
 in
 {
-    assert(dp.getName == departureNodeName);
+    assert(dp.name == departureNodeName);
 }
 body
```

Comment:
*ahem* ;-)

Reply:
Will address this before merging, was an oversight by me...